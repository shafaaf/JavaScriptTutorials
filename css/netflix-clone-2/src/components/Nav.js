import './Nav.css';

export default function Nav() {
    return (
        <div className="nav">
            <img
                className="nav_logo"
                src="https://cdn.vox-cdn.com/thumbor/QuS2QKQys3HhosKiV-2IuKhphbo=/39x0:3111x2048/1400x1050/filters:focal(39x0:3111x2048):format(png)/cdn.vox-cdn.com/uploads/chorus_image/image/49901753/netflixlogo.0.0.png"
                alt="Netflix Logo"
            />

            <img
                className="nav_avatar"
                src="https://upload.wikimedia.org/wikipedia/commons/0/0b/Netflix-avatar.png"
                alt="Netflix Avatar"
            />

        </div>
    )
}
