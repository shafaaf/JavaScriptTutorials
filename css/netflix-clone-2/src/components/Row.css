.row {
    margin-left: 20px;
}

.row_name {
 color: white;
}

.row_posters {
    display: flex;
    overflow-y: hidden;
    overflow-x: scroll;
    padding: 40px;
}

/* Dont show scroll bars*/
.row_posters::-webkit-scrollbar {
    display: none;
}

.row_poster {
    object-fit: contain;
    width: 100%;
    max-height: 200px;
    margin-right: 10px;
    transition: transform 200ms;
}

.row_poster:hover {
    transform: scale(1.2);
}

.row_posterLarge {
    max-height: 350px;
}
