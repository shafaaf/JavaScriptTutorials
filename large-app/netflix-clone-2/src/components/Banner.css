.banner {
    color: white;
    object-fit: contain;
    height: 448px;
}
.banner_contents {
    margin-left: 101px;
    padding-top: 140px;
    height: 190px;
}
.banner_title {
    font-size: 3em;
    font-weight: 800;
    padding-bottom: 0.3rem;
}

.banner_description {
    width: 45rem;
    line-height: 1.7;
    padding-top: 2rem;
    font-size: 1rem;
    max-width: 560px;
    height: 80px;
}

.banner_button {
    cursor: pointer;
    color: white;
    outline: none;
    border: none;
    font-weight: 700;
    border-radius: 0.2vw;
    padding-left: 2rem;
    padding-right: 2rem;
    margin-top: 2rem;
    margin-right: 1rem;
    padding-top: 0.5rem;
    padding-bottom: 0.5rem;
    background-color: rgba(51, 51, 51, 0.5);
}