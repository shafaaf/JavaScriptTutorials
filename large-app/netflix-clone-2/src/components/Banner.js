import React, {useState, useEffect} from 'react';
import axiosInstance from "../axios";
import requests from "../requests";

function Banner() {
    const [movie, setMovie] = useState([]);

    useEffect(() => {
        async function fetchData () {
            const request = await axiosInstance.get(requests.fetchNetflixOriginals);
            const randomIndex = Math.floor(Math.random() * request.data.results.length);
            setMovie (request.data.results[randomIndex]);
            return request;
        }
        fetchData().then(() => console.log(movie));
    }, []);

    return (
        <div>
            <header
                className="banner"
                style={{
                    backgroundSize: "cover",
                    backgroundImage: `url(https://www.irmi.com/images/default-source/article-images/aviation/boeing-737.jpg?sfvrsn=4)`,
                    backgroundPosition: "center center"
                }}

            >
                <div className="banner_contents">

                </div>
            </header>

        </div>
    );
}

export default Banner;