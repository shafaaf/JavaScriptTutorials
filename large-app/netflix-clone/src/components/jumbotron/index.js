import React from "react";
import Inner from './styles/jumbotron';

const Jumbotron = () => {
    return <Inner>I am the Jumbotron</Inner>
};

export default Jumbotron;
