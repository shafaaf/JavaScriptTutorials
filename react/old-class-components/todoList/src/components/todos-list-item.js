// Bind this used so that 'this' object can be used
// within the method

// See how CSS colors styles are applied

// Need to change state for tasks when clicked and
// so need a prop which modifies state of previous component

import React from 'react';

export default class  TodosListItem extends React.Component{
	constructor(props){
		super(props);
		this.state = {
			isEditing: false
		};
	}

	// Change editing state to true on edit click
	onEditClick() {
		console.log("Editing");
		this.setState ({
			isEditing: true
		});
	}

	// Change cancel state to false on canel click
	onCancelClick(){
		console.log("Cancelling");
		this.setState ({
			isEditing: false
		});
	}

	onSaveClick(event){
		event.preventDefault();
		var oldTask = this.props.task;
		var newTask = this.refs.editInput.value;
		console.log("onSaveClick: oldTask is: ", oldTask);
		console.log("onSaveClick: newTask is: ", newTask);
		this.props.saveTask(oldTask, newTask);
		this.setState({isEditing: false});
	}

	renderTaskSection() {
		//console.log("this.props is: ", this.props);
		const task = this.props.task;
		const isCompleted = this.props.isCompleted;
		const taskStyle = {
			color: isCompleted ? 'green': 'red',
			cursor: 'pointer'
		};

		if (this.state.isEditing) { // Editing so need input box
			return (
				<td>
					<form onSubmit = {this.onSaveClick.bind(this)}>
						<input type = "text" placeholder = {task}
							ref = "editInput"/>
					</form>
				</td>
			);
		}
		else { // Not editing so just return the task name
			return (
				<td style = {taskStyle} onClick = {this.props.toggleTask.bind(this, task)}>
					{task}
				</td>
			);
		}
	}

	renderActionSection() {
		if(this.state.isEditing) {
			return ( // Return buttons (save, cancel) when editing
				<td>
					<button onClick = {this.onSaveClick.bind(this)}>Save</button>
					<button onClick = {this.onCancelClick.bind(this)}>Cancel</button>
				</td>
			);
		}
		else { // Return default buttons (edit, delete) since editing
			return(
				<td>
					<button onClick = {this.onEditClick.bind(this)}>Edit</button>
					<button onClick = {this.props.deleteTask.bind(this, this.props.task)}>Delete</button>
				</td>
			);
		}
	}

	render() {
		return (
			<tr>
				{this.renderTaskSection()}
				{this.renderActionSection()}
			</tr>
		);
	}
}
