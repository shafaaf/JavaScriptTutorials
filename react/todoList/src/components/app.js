// this.setState always re renders the page at the end

// pass states using props and if they change call
// this.prop.propName to call the parent component's
// function which changes state. This parent component
// should be the one that declares the state.

// play with "this" keyword in toggleTask. The toggle task in todos-list-item
// also has a this binded to pass the task
// If remove the "this" in this file, this refers to TodosListItem

import React from 'react';
import CreateTodo from './create-todo';
import TodosList from './todos-list';

const todos = [
	{
		task: 'make react tutorial',
		isCompleted: false
	},
	{
		task: 'eat dinner',
		isCompleted: true
	},
];

export default class App extends React.Component {

	constructor(props){
		super(props);
		this.state = {
			todos: todos
		}
	}

	// Called when new task submitted in form
	createTask(task) {
		// console.log("createTask with task: ", task);
		// console.log("this is: ", this);
		this.state.todos.push({
			task: task,
			isCompleted: false
		});
		this.setState({
			todos: this.state.todos
		});
	}

	// Called when a task is clicked on
	toggleTask(task) {
		console.log("At toggleTask. Task is: ", task);
		console.log("This is: ", this);
		const todos = this.state.todos;
		const todosLength = todos.length;
		for (let i = 0; i < todosLength; i++) {
			if (todos[i]["task"] === task){
				const isCompleted = todos[i]["isCompleted"];
				todos[i]["isCompleted"] = !isCompleted;
				break;
			}
		}
		this.setState({ todos: this.state.todos});
	}

	// rename a task
	saveTask(oldTask, newTask) {
		const todos = this.state.todos;
		const todosLength = todos.length;
		for (let i = 0; i < todosLength; i++) {
			if (todos[i]["task"] === oldTask) {
				todos[i]["task"] = newTask;
				break;
			}
		}
		this.setState({ todos: this.state.todos});
	}

	deleteTask(taskToDelete) {
		const todos = this.state.todos;
		const todosLength = todos.length;
		for (let i = 0; i < todosLength; i++) {
			if (todos[i]["task"] === taskToDelete){
				todos.splice(i,1);
				break;
			}
		}
		this.setState({ todos: this.state.todos});
	}

	render() {
		return (
			<div>
				<h1>React todo app</h1>
				<CreateTodo todos = {this.state.todos} createTask = {this.createTask.bind(this)}/>
				<TodosList
					todos = {this.state.todos}
					toggleTask = {this.toggleTask.bind(this)}
					saveTask = {this.saveTask.bind(this)}
					deleteTask = {this.deleteTask.bind(this)}/>
			</div>
		);
	}
}
