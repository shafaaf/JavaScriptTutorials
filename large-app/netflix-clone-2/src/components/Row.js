import React from 'react';
import { useEffect, useState } from "react";
import axiosInstance from "../axios";
import './Row.css';

const baseUrl = 'https://image.tmdb.org/t/p/original/';

const Row = ({title, fetchUrl, isLargeRow}) => {
    const [movies, setMovies] = useState([]);

    useEffect(() => {
        async function fetchData () {
            const request = await axiosInstance.get(fetchUrl);
            setMovies(request.data.results);
        }
        fetchData();

    }, [fetchUrl]);
    return (
        <div className="row">
            <h2>{title}</h2>
            <div className="row_posters">
                {
                    movies.map(movie => {
                        return <img key = {movie.id}
                            className={`row_poster ${isLargeRow && "row_posterLarge"}`}
                            src={`${baseUrl}${isLargeRow? movie.poster_path: movie.backdrop_path }`}
                            alt={movie.name}
                        />
                    })
                }
            </div>
        </div>
    );
};

export default Row;
