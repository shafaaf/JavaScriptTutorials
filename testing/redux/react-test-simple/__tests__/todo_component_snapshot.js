import React from 'react';
import toJson from 'enzyme-to-json';
import moment from 'moment';
import { shallow } from 'enzyme';
import { Todo } from '../src/components/Todo';

it('Renders correctly', () => {
  const wrapper = shallow(
    <Todo
      id="1"
      title="Todo"
      project="Project"
      done={false}
      url="https://www.photos.com/a_photo"
      createdAt={moment().subtract(1, 'days').format()}
      editTodo={jest.fn()}
      toggleTodo={jest.fn()}
      deleteTodo={jest.fn()}
    />,
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});
