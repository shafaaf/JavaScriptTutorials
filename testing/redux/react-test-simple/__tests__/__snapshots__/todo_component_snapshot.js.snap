// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className="column"
>
  <div
    className="ui brown card"
  >
    <img
      className="ui image"
      src="https://www.photos.com/a_photo"
    />
    <div
      className="content"
    >
      <div
        className="header"
      >
        Todo
      </div>
      <div
        className="meta"
      >
        Project
      </div>
      <div
        className="meta"
      >
        Created 
        a day ago
      </div>
    </div>
    <div
      className="extra content"
    >
      <div>
        <div
          className="ui toggle checkbox"
          style={
            Object {
              "marginBottom": "10px",
            }
          }
        >
          <input
            defaultChecked={false}
            name="public"
            onChange={[Function]}
            type="checkbox"
            value="on"
          />
          <label>
            Complete
          </label>
        </div>
        <div
          className="ui two buttons"
        >
          <button
            className="ui basic green button"
            onClick={[Function]}
          >
            Edit
          </button>
          <button
            className="ui basic red button"
            onClick={[Function]}
          >
            Delete
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
